studyBuddy/src/main/java/com/google/sps/models/Student.java

```diff
@@ -0,0 +1,19 @@
+// This is a Student class model 
+
+package com.google.sps.models;
+
+public final class Student{
+    private String firstName;
+    private String lastName;
+    private String school;
+    private String subject;
+    private long timestamp;
+
+    public Student(String firstName, String lastName, String school, String subject, long timestamp){
```

Comment:
looks gr8, just wanted to clarify that the login info would fill most of this out and so when we search we essentially just take the subject to fill the rest? but going to approve (:

Reply:
Hi @sgh005 

Yes, that is right. The Authentication feature will fill in the following:

- ```email``` (which I forgot to add to the class model but lol)
- ```last name``` (will be part of the email I think)
- ```school``` (will be part of the email for sure)

If we want to collect the first name as well, we may need an additional **Sign Up Page** where we prompt for the ```first name``` (or simply a username/nickname) or something of that sort because the authentication library only gives us access to the user's **email address**. 

But for now the client is only sending the ```subject```. 

Thank for the review :) 

Ben

Reply:
Makes sense! No yeah that works I was just making sure I understood our implementation. Also for whatever reason I am unable to approve the request as it needs Michaels approval. 
-Steph

Reply:
Having a first and last name would be a good feature, which we might be able to add once we have the idea of a user profile. But for now, I think we will want to just use the email address as the identifier because it will be a unique to each user and doesn't require us setting up any additional inputs.

As for the whole last name thing, I don't think it is standard to use last names as emails (my school doesn't, because there are too many students with the same last name).

Reply:
Hi @mpasetes17 

Understood! 

- I built the Entity based on the user sample that is on Search Page.  
  If you look closely I hardcoded the **first name: John ** **last name: Doe** 
 
- Give me a suggestion here: Should I get rid of the first and last name properties for now or should I just leave them there but hardcoded to illustrate a potential integration of the feature?  
  
- Thanks for letting me know that last names in school emails are not standard. I will keep that in mind. 


Ben

Reply:
I don't think there is any harm in leaving them in there, especially since this will be something that we would like to implement fairly soon after we get the MVP. I would say go ahead and leave the field, but hard code the value so that it will be easily distinguishable as not valid data about the entry. I don't know how exactly we want to lay out the data store table, but this could be as simple as setting both to NULL until we are able to implement that feature.